src/main/java/juja/microservices/gamification/service/AchievementService.java

```diff
@@ -70,4 +73,40 @@ public String addDaily(String userFromId, String description) {
         }
         return result;
     }
+
+    public List<String> addCodenjoy(String  userFromId, String firstUserToId, String secondUserToId,
+            String thirdUserToId) {
+        List<String> result = new ArrayList<>();
+        List<Achievement> codenjoyUsersToday = achievementRepository.getAllCodenjoyAchievementsCurrentDate();
+        if (!codenjoyUsersToday.isEmpty()) {
+            throw new UnsupportedAchievementException("You cannot give codenjoy points twice a day");
+        } else if ("".equals(userFromId)) {
```

Comment:
I think the logic about required fuilds should be checked on the slack-bot level, before the request went to the server side. We can discuss it on meeting.